Tests/Unit/DependencyInjection/Compiler/DescriptionEnhancerPassTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2016 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\ResourceBundle\Tests\Unit\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Cmf\Bundle\ResourceBundle\DependencyInjection\Compiler\DescriptionEnhancerPass;
+use Symfony\Component\DependencyInjection\Definition;
+use Prophecy\Argument;
+use Symfony\Component\DependencyInjection\Reference;
+
+class DescriptionEnhancerPassTest extends \PHPUnit_Framework_TestCase
+{
+    public function setUp()
+    {
+        $this->container = $this->prophesize(ContainerBuilder::class);
+        $this->factoryDefinition = $this->prophesize(Definition::class);
+        $this->pass = new DescriptionEnhancerPass();
+    }
+
+    /**
+     * It should return early if the factory does not exist.
+     *
+     * TODO: WHY????
+     */
+    public function testReturnEarlyFactoryNotExist()
+    {
+        $this->container->has(
+            'cmf_resource.description.factory'
+        )->willReturn(false);
+        $this->pass->process($this->container->reveal());
+        $this->container->getParameter(Argument::cetera())->shouldNotHaveBeenCalled();
```

Comment:
I would prefer to speciffy all promises and predictions first and then execute it.